src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/validator/AppointmentValidator.kt

```diff
@@ -36,6 +38,18 @@ class AppointmentValidator {
         }
       }
     }
+
+    if (referralStartDate != null && updateAppointmentDTO.appointmentTime.isBefore(referralStartDate)) {
+      errors.add(FieldError(field = "appointmentTime", error = Code.INVALID_VALUE))
```

Comment:
Its a general doubt, are we going to throw only code.INVALID_VALUE and not anything specific like "appointmentTime is before the referral start date"